mitodl/bootcamp-ecommerce
# Allow users to update resume until submission is reviewed

#### What are the relevant tickets?
Closes #960 

#### What's this PR do?
- Modifies state transition criteria, allowing users to update their resume until submissions have been reviewed.
- Modifies the front-end to hide the View/edit/upload resume link if the application is not in an appropriate state.

#### How should this be manually tested?
- Start a new application.  Upload a resume or set a linked in profile.  It should work and the application state should transition to `AWAITING_USER_SUBMISSIONS`
- Change the resume or linked in profile.  It should work and the application state should remain `AWAITING_USER_SUBMISSIONS`
- Either complete a video interview or artificially create one via django admin - make sure submission status is 'submitted, application state is `AWAITING_SUBMISSION_REVIEW`.
- Refresh the application details.  Change the resume or linked in profile again.  It should work and the application state should remain `AWAITING_SUBMISSION_REVIEW`.
- As an admin user, approve or reject the submission from the `/submissions` URL.
- Refresh the application details.  There should no longer be a link to view/update your resume.
